### DISCUSSION TITLE
DB Rollup Migration?

### DISCUSSION THREAD

**Discussion by andre-qumulo** [sqlalchemy/alembic#1476] [Usage Questions] [answered]
The way we run unit tests against our code is to create a new DB against an in-memory/temporary file system and then apply the migrations.

As we have added more migrations our test time has increased.

I was wondering if there was a way to produce a rollup migration that would create the schema for the current models if it doesn't exist?

**Comment by zzzeek** [comment, marked as answer]
if you are testing the migrations, then I assume you want to run them all for your test suite.  if you are not testing the migrations, then I'd just use [create_all()](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.MetaData.create_all) for your test suite's purposes.       

As far a "rollup migration" you can do this also but you'd need to construct the migration code by hand, or otherwise run an autogenerate against a database which is updated to the target version for the rollup, removing all migration files above that version so that autogenerate thinks it's at the head (you can make a new migration directory for this kind of thing).

**Reply by andre-qumulo** [reply]
Interesting.  I think that's what I'm looking for, thank you!
